backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceBreadcrumbs.java

```diff
@@ -0,0 +1,275 @@
+     */
+    public void addBreadcrumb(String message, Map<String, Object> attributes) {
+        addBreadcrumb(message, attributes, BacktraceBreadcrumbType.MANUAL, BacktraceBreadcrumbLevel.INFO);
+    }
+
+    /**
+     * Add a breadcrumb of type "Manual" and the desired level with the provided message string and attributes
+     * @param message
+     * @param attributes
+     * @param level
+     */
+    public void addBreadcrumb(String message, Map<String, Object> attributes, BacktraceBreadcrumbLevel level) {
+        addBreadcrumb(message, attributes, BacktraceBreadcrumbType.MANUAL, level);
+    }
+
+    /**
+     * Add a breadcrumb of the desired type and level "Info" with the provided message string
+     * @param message
+     * @param type
+     */
+    public void addBreadcrumb(String message, BacktraceBreadcrumbType type) {
+        addBreadcrumb(message, null, type, BacktraceBreadcrumbLevel.INFO);
+    }
+
+    /**
+     * Add a breadcrumb of the desired level and type with the provided message string
+     * @param message
+     * @param type
+     * @param level
+     */
+    public void addBreadcrumb(String message, BacktraceBreadcrumbType type, BacktraceBreadcrumbLevel level) {
+        addBreadcrumb(message, null, type, level);
+    }
+
+    /**
+     * Add a breadcrumb of the desired type and level "Info" with the provided message string and attributes
+     * @param message
+     * @param attributes
+     * @param type
+     */
+    public void addBreadcrumb(String message, Map<String, Object> attributes, BacktraceBreadcrumbType type) {
+        addBreadcrumb(message, attributes, type, BacktraceBreadcrumbLevel.INFO);
+    }
+
+    /**
+     * Add a breadcrumb of the desired level and type with the provided message string and attributes
+     * @param message
+     * @param attributes
+     * @param type
+     * @param level
+     */
+    public void addBreadcrumb(String message, Map<String, Object> attributes, BacktraceBreadcrumbType type, BacktraceBreadcrumbLevel level) {
+        // We use currentTimeMillis in the BacktraceReport too, so for consistency
+        // we will use it here.
+        long time = System.currentTimeMillis();
+
+        String serializedAttributes = "{}";
+        if (attributes != null) {
+            Gson gson = new Gson();
+            Type gsonType = new TypeToken<HashMap>(){}.getType();
+            serializedAttributes = gson.toJson(attributes,gsonType);
+        }
+
+        addBreadcrumb(time,
+                type.ordinal(),
+                level.ordinal(),
+                message,
+                serializedAttributes);
+    }
+
+    // Create a breadcrumb which reflects the current breadcrumb configuration
+    private void addConfigurationBreadcrumb()
+    {
+        Map<String, Object> attributes = new HashMap<String, Object>();
+
+        for (BacktraceBreadcrumbType enabledType : BacktraceBreadcrumbType.values())
+        {
+            if (enabledBreadcrumbTypes != null &&
```

Comment:
If you will write it in this way it will be more concise
```
String state = (enabledBreadcrumbTypes != null &&
                    enabledBreadcrumbTypes.contains(enabledType)) ? "enabled" : "disabled";

attributes.put(enabledType.name(), state);
```

Reply:
Nice, will make this change!